service/history/historyEngine2_test.go

```diff
@@ -142,25 +143,39 @@ func (s *engine2Suite) SetupTest() {
 		eventsCache:               s.mockEventsCache,
 		timeSource:                clock.NewRealTimeSource(),
 	}
+	s.mockClusterMetadata.On("IsGlobalDomainEnabled").Return(false)
+	s.mockClusterMetadata.On("GetCurrentClusterName").Return(cluster.TestCurrentClusterName)
+	s.mockClusterMetadata.On("ClusterNameForFailoverVersion", common.EmptyVersion).Return(cluster.TestCurrentClusterName)
+	s.mockTxProcessor = &MockTransferQueueProcessor{}
+	s.mockTxProcessor.On("NotifyNewTask", mock.Anything, mock.Anything).Maybe()
+	s.mockReplicationProcessor = &mockQueueProcessor{}
+	s.mockReplicationProcessor.On("notifyNewTask").Maybe()
+	s.mockTimerProcessor = &MockTimerQueueProcessor{}
+	s.mockTimerProcessor.On("NotifyNewTimers", mock.Anything, mock.Anything).Maybe()
 
 	historyCache := newHistoryCache(mockShard)
 	h := &historyEngineImpl{
-		currentClusterName: mockShard.GetService().GetClusterMetadata().GetCurrentClusterName(),
-		shard:              mockShard,
-		executionManager:   s.mockExecutionMgr,
-		historyMgr:         s.mockHistoryMgr,
-		historyV2Mgr:       s.mockHistoryV2Mgr,
-		historyCache:       historyCache,
-		logger:             s.logger,
-		metricsClient:      metrics.NewClient(tally.NoopScope, metrics.History),
-		tokenSerializer:    common.NewJSONTaskTokenSerializer(),
-		config:             s.config,
-		archivalClient:     s.mockArchivalClient,
-		timeSource:         mockShard.timeSource,
+		currentClusterName:   mockShard.GetService().GetClusterMetadata().GetCurrentClusterName(),
+		shard:                mockShard,
+		clusterMetadata:      s.mockClusterMetadata,
+		executionManager:     s.mockExecutionMgr,
+		historyMgr:           s.mockHistoryMgr,
+		historyV2Mgr:         s.mockHistoryV2Mgr,
+		historyCache:         historyCache,
+		logger:               s.logger,
+		metricsClient:        metrics.NewClient(tally.NoopScope, metrics.History),
+		tokenSerializer:      common.NewJSONTaskTokenSerializer(),
+		config:               s.config,
+		archivalClient:       s.mockArchivalClient,
+		timeSource:           mockShard.timeSource,
+		historyEventNotifier: newHistoryEventNotifier(clock.NewRealTimeSource(), metrics.NewClient(tally.NoopScope, metrics.History), func(string) int { return 0 }),
```

Comment:
Would be better to reuse metricsClient, and declare the last arg outside for better understanding.